modules/python/python-grammar.ym

```diff
@@ -72,11 +72,13 @@ start
         | LL_CONTEXT_SOURCE KW_PYTHON
           {
             last_driver = *instance = python_sd_new(configuration);
+            g_assert(!((LogPipe *)last_driver)->generate_persist_name);
```

Comment:
Grammar error would not be suitable here. This is not a user error, but a syslog-ng error. It the users runs into this it can only be resolved from c code. That's why I used `g_assert` here.

Reply:
Without understanding the patch in details, does that happen that this is NULL in this case? Because we are not usually checking the nullness of any methods that are otherwise required for a feature to operate.

So again without understanding the details this assertion shouldn't be needed here.

Reply:
Unfortunately this is an operational assert.

We almost everytime set the generate_persist_name in the new operation, that, as it turned out causes backward compatibility problem.

This is the first set of work of a larger feature: to support persist and message acknowledgement in python source. The problem is, when python source was introduced, we did not specify persist name. This means persist name collision detection is currently switched off for python sources. But for persist api to work, I will need a persist name. 

If I added a default persist name (setting generate_persist_name in the new method), that would mean those users that reuse the same python class multiple times, their configuration after update would not start due to persist collision.

To avoid that, I need to set generate_persist_name conditionally, and do that during init, only after the python instance is read and initialized (`@staticmethod generate_persist_name_method` is collected).

This is something that is easy to miss, at least I ran into it. To avoid that someone tries to do a refactor and move generate_persist_name setting to new from init, I added this assert.